src/tsd.c

```diff
@@ -280,6 +280,10 @@ tsd_fetch_slow(tsd_t *tsd, bool minimal) {
 		tsd_slow_update(tsd);
 	} else if (tsd_state_get(tsd) == tsd_state_uninitialized) {
 		if (!minimal) {
+			if (!tsd_booted) {
```

Comment:
Given that this is the only early return in the function, IMO this would read better if the following code (lines 287-291) were instead wrapped in `if (tsd_booted) { }`.